### PR TITLE
ARROW-14937: [Doc] Make sure the docs directory is mounted as a volume

### PR DESCRIPTION
**Pull request by amol-**

### CODE REVIEW

File: docs/source/developers/documentation.rst [hunk 1]
```diff
@@ -94,7 +94,7 @@ Docker container.
 
 .. code-block:: shell
 
-  archery docker run ubuntu-docs
+  archery docker run -v "$(PWD)/docs:/build/docs" ubuntu-docs
 
 The final output is located under ``docs/_build/html``.
```
Comment (kszucs) [inline docs/source/developers/documentation.rst:99]: @amol- This may not be valid due to the mount. Can you confirm?

Comment (amol-) [reply docs/source/developers/documentation.rst:99]: Uhm, not sure I get what you mean. If you are talking about the final output location, once the `-v` option is provided it's correct as we are building the local `docs` directory as `/build/docs` where the actual output is written.

Comment (kszucs) [reply docs/source/developers/documentation.rst]: I've just fixed the build script to generate the docs within the container directly to `/build/docs`.

File: docs/source/developers/documentation.rst [hunk 2]
```diff
@@ -94,9 +94,9 @@ Docker container.
 
 .. code-block:: shell
 
-  archery docker run ubuntu-docs
+  archery docker run -v "$(PWD)/docs:/build/docs" ubuntu-docs
```
Comment (kou) [inline docs/source/developers/documentation.rst:97]:
```suggestion
  archery docker run -v "${PWD}/docs:/build/docs" ubuntu-docs
```

File: docs/source/developers/documentation.rst [hunk 3]
```diff
@@ -94,9 +94,9 @@ Docker container.
...
 
-The final output is located under ``docs/_build/html``.
+The final output is located under the ``$(PWD)/docs`` directory.
```
Comment (kou) [inline docs/source/developers/documentation.rst:99]:
```suggestion
The final output is located under the ``${PWD}/docs`` directory.
```

**Review by kszucs — APPROVED** [review]

**Pull request closed without merging**